tests/models/detr/test_modeling_detr.py

```diff
@@ -585,13 +585,35 @@ def test_inference_no_head(self):
 
         expected_shape = torch.Size((1, 100, 256))
         assert outputs.last_hidden_state.shape == expected_shape
-        expected_slice = torch.tensor(
-            [
-                [0.0622, -0.5142, -0.4034],
-                [-0.7628, -0.4935, -1.7153],
-                [-0.4751, -0.6386, -0.7818],
-            ]
-        ).to(torch_device)
+        expected_slices = Expectations(
+            {
+                ("xpu", 3): torch.tensor(
+                    [
+                        [0.0622, -0.5142, -0.4034],
+                        [-0.7628, -0.4935, -1.7153],
+                        [-0.4751, -0.6386, -0.7818],
+                    ], device=torch_device),
+                ("cuda", 7): torch.tensor(
+                    [
+                        [0.0622, -0.5142, -0.4034],
+                        [-0.7628, -0.4935, -1.7153],
+                        [-0.4751, -0.6386, -0.7818],
+                    ], device=torch_device),
+                ("cuda", 8): torch.tensor(
+                    [
+                        [0.0622, -0.5142, -0.4034],
+                        [-0.7628, -0.4935, -1.7153],
+                        [-0.4751, -0.6386, -0.7818],
+                    ], device=torch_device),
+                ("rocm", (9, 5)): torch.tensor(
+                    [
+                        [ 0.0616, -0.5146, -0.4032],
+                        [-0.7629, -0.4934, -1.7153],
+                        [-0.4768, -0.6403, -0.7826]
+                    ], device=torch_device),
+            }
+        )  # fmt: skip
```

Comment:
It would be cleaner to: 
1. have only lists in the expectations dict, and create a single on-device tensor from the expected list
2. have only the `rocm` expectation and a `Default` expectation to cover the 3 repeating ones